idl/github.com/uber/cadence/shared.thrift

```diff
@@ -962,6 +962,22 @@ struct SignalWorkflowExecutionRequest {
   70: optional binary control
 }
 
+struct SignalWithStartWorkflowExecutionRequest {
```

Comment:
out of curiosity, why not just add the signal part to existing start workflow API?

Reply:
That's a decision after design meeting. We prefer creating new API (and related requests) for such new cases. e.g. completeActivityByID is new instead of reuse. 
But we still try best to reuse as much as possible.